protoc-gen-swagger/genswagger/template.go

```diff
@@ -1448,6 +1459,88 @@ func processExtensions(inputExts map[string]*structpb.Value) ([]extension, error
 	return exts, nil
 }
 
+func validateHeaderType(headerType string) error {
+	switch headerType {
+	case
+		"string",
+		"number",
+		"integer",
+		"boolean":
+		return nil
```

Comment:
Could we add a comment with a reference to the relevant part of the spec laying out the permitted values here (and why we only allow a subset)?

Reply:
Done
```
	// The type of the object. The value MUST be one of "string", "number", "integer", "boolean", or "array"
	// See: https://github.com/OAI/OpenAPI-Specification/blob/3.0.0/versions/2.0.md#headerObject
	// Note: currently not implementing array as we are only implementing this in the operation response context
```